collect_app/src/main/java/org/odk/collect/android/database/DatabaseConnection.kt

```diff
@@ -58,6 +58,22 @@ open class DatabaseConnection(
         private val openHelpers = mutableMapOf<String, SQLiteOpenHelper>()
         private val toClose = mutableListOf<SQLiteOpenHelper>()
 
+        @JvmStatic
+        fun close(projectId: String) {
```

Comment:
I'm not a huge fan of that this method cares about the project ID, as it doesn't really know (or doesn't need to know) anything about projects. Also, given the implementation, this parameter could really just be called `subDirectory` as if my db was in `/dir1/dir2/projectDir/` I coud pass `dir1` or `dir2` into this and it would close (and clear) the connection.

I'm wondering if an alternative approach would be to have a `cleanUp` method that closes and clears connections that are to no longer existing fils (sort of like we do just-in-time in when grabbing a db helper)? We could call this after deleting the project dir and it should then remove the helpers we want, without us needing to care about any identifiers in `DatabaseConnection`.

Reply:
So it's a similar solution it just check the file existence instead of finding it by project id. Updated, please take a look if it's what you expected.